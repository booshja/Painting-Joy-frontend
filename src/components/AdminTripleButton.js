// dependencies
import React from "react";
import styled from "styled-components";
// hooks
import { useHistory } from "react-router";

const StyledButtonContainer = styled.div`
    display: flex;
    align-items: center;
    margin-top: 1rem;
    align-self: center;
`;

const StyledLeftButton = styled.button`
    border: 2px solid #207070;
    border-radius: 4px 0 0 4px;
    background-color: #207070;
    color: #ffffff;
    font-family: "News Cycle", sans-serif;
    font-size: 1rem;
    font-weight: 700;
    letter-spacing: 1px;
    padding: 2.5px 24px;
`;

const StyledMiddleButton = styled.button`
    border: none;
    border-top: 2px solid #207070;
    border-bottom: 2px solid #207070;
    color: #207070;
    background-color: #ffffff;
    font-family: "News Cycle", sans-serif;
    font-size: 1rem;
    font-weight: 700;
    letter-spacing: 1px;
    padding: 2.5px 24px;
`;

const StyledRightButton = styled.button`
    border: 2px solid #db9487;
    background-color: #db9487;
    color: #ffffff;
    border-radius: 0 4px 4px 0;
    font-size: 1rem;
    font-weight: 700;
    letter-spacing: 1px;
    padding: 2.5px 24px;
    font-family: "News Cycle", sans-serif;
`;

const AdminTripleButton = ({
    id,
    handleArchive,
    handleUnArchive,
    handleDelete,
    showArchived,
}) => {
    // set up history
    const history = useHistory();

    return (
        <StyledButtonContainer>
            <StyledLeftButton
                onClick={() => history.push(`/admin/murals/edit/${id}`)}
            >
                Edit
            </StyledLeftButton>
            <StyledMiddleButton
                onClick={() =>
                    showArchived ? handleUnArchive(id) : handleArchive(id)
                }
            >
                {showArchived ? "Unarchive" : "Archive"}
            </StyledMiddleButton>
            <StyledRightButton onClick={() => handleDelete(id)}>
                Delete
            </StyledRightButton>
        </StyledButtonContainer>
    );
};

export default AdminTripleButton;
