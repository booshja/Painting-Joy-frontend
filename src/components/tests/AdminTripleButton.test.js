import { getAllByRole, render } from "@testing-library/react";
import AdminTripleButton from "../AdminTripleButton";
import "jest-styled-components";

/** Smoke Test */
it("renders without crashing, showArchived true", () => {
    render(
        <AdminTripleButton
            id={1}
            handleArchive={null}
            handleUnArchive={null}
            handleDelete={null}
            showArchived={true}
        />
    );
});

it("renders without crashing, showArchived false", () => {
    render(
        <AdminTripleButton
            id={1}
            handleArchive={null}
            handleUnArchive={null}
            handleDelete={null}
            showArchived={false}
        />
    );
});

/** Snapshot Test */
it("matches snapshot, showArchived true", () => {
    const { asFragment } = render(
        <AdminTripleButton
            id={1}
            handleArchive={null}
            handleUnArchive={null}
            handleDelete={null}
            showArchived={true}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, showArchived false", () => {
    const { asFragment } = render(
        <AdminTripleButton
            id={1}
            handleArchive={null}
            handleUnArchive={null}
            handleDelete={null}
            showArchived={false}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly, showArchived true", () => {
    const { getAllByRole } = render(
        <AdminTripleButton
            id={1}
            handleArchive={null}
            handleUnArchive={null}
            handleDelete={null}
            showArchived={true}
        />
    );

    const buttons = getAllByRole("button");
    expect(buttons[0]).toHaveTextContent("Edit");
    expect(buttons[1]).toHaveTextContent("Unarchive");
    expect(buttons[2]).toHaveTextContent("Delete");
});

it("renders correctly, showArchived false", () => {
    const { getAllByRole } = render(
        <AdminTripleButton
            id={1}
            handleArchive={null}
            handleUnArchive={null}
            handleDelete={null}
            showArchived={false}
        />
    );

    const buttons = getAllByRole("button");
    expect(buttons[0]).toHaveTextContent("Edit");
    expect(buttons[1]).toHaveTextContent("Archive");
    expect(buttons[2]).toHaveTextContent("Delete");
});
