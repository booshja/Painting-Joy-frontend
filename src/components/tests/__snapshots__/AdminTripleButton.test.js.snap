// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, showArchived false 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 1rem;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c1 {
  border: 2px solid #207070;
  border-radius: 4px 0 0 4px;
  background-color: #207070;
  color: #ffffff;
  font-family: "News Cycle",sans-serif;
  font-size: 1rem;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 2.5px 24px;
  cursor: pointer;
}

.c1:hover {
  color: #207070;
  background-color: #ffffff;
}

.c2 {
  border: none;
  border-top: 2px solid #207070;
  border-bottom: 2px solid #207070;
  color: #207070;
  background-color: #ffffff;
  font-family: "News Cycle",sans-serif;
  font-size: 1rem;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 2.5px 24px;
  cursor: pointer;
}

.c2:hover {
  color: #ffffff;
  background-color: #207070;
}

.c3 {
  border: 2px solid #db9487;
  background-color: #db9487;
  color: #ffffff;
  border-radius: 0 4px 4px 0;
  font-size: 1rem;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 2.5px 24px;
  font-family: "News Cycle",sans-serif;
  cursor: pointer;
}

.c3:hover {
  color: #db9487;
  background-color: #ffffff;
}

@media screen and (min-width:1024px) {
  .c0 {
    -webkit-align-self: flex-start;
    -ms-flex-item-align: start;
    align-self: flex-start;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    margin-left: 2rem;
  }
}

<div
    class="c0"
  >
    <button
      class="c1"
    >
      Edit
    </button>
    <button
      class="c2"
    >
      Archive
    </button>
    <button
      class="c3"
    >
      Delete
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot, showArchived true 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 1rem;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c1 {
  border: 2px solid #207070;
  border-radius: 4px 0 0 4px;
  background-color: #207070;
  color: #ffffff;
  font-family: "News Cycle",sans-serif;
  font-size: 1rem;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 2.5px 24px;
  cursor: pointer;
}

.c1:hover {
  color: #207070;
  background-color: #ffffff;
}

.c2 {
  border: none;
  border-top: 2px solid #207070;
  border-bottom: 2px solid #207070;
  color: #207070;
  background-color: #ffffff;
  font-family: "News Cycle",sans-serif;
  font-size: 1rem;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 2.5px 24px;
  cursor: pointer;
}

.c2:hover {
  color: #ffffff;
  background-color: #207070;
}

.c3 {
  border: 2px solid #db9487;
  background-color: #db9487;
  color: #ffffff;
  border-radius: 0 4px 4px 0;
  font-size: 1rem;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 2.5px 24px;
  font-family: "News Cycle",sans-serif;
  cursor: pointer;
}

.c3:hover {
  color: #db9487;
  background-color: #ffffff;
}

@media screen and (min-width:1024px) {
  .c0 {
    -webkit-align-self: flex-start;
    -ms-flex-item-align: start;
    align-self: flex-start;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    margin-left: 2rem;
  }
}

<div
    class="c0"
  >
    <button
      class="c1"
    >
      Edit
    </button>
    <button
      class="c2"
    >
      Unarchive
    </button>
    <button
      class="c3"
    >
      Delete
    </button>
  </div>
</DocumentFragment>
`;
